core/src/server/rpc_module.rs

```diff
@@ -569,14 +569,15 @@ impl<Context: Send + Sync + 'static> RpcModule<Context> {
 	}
 
 	/// Register a new asynchronous RPC method, which computes the response with the given callback.
-	pub fn register_async_method<R, Fun, Fut>(
+	pub fn register_async_method<R, E, Fun, Fut>(
 		&mut self,
 		method_name: &'static str,
 		callback: Fun,
 	) -> Result<MethodResourcesBuilder, Error>
 	where
 		R: Serialize + Send + Sync + 'static,
-		Fut: Future<Output = Result<R, Error>> + Send,
+		E: Into<Error>,
```

Comment:
It didn't work with `E: Into<ErrorObjectOwned>`? 

Would be great skip the conversion to `jsonrpsee::core::Error`

Reply:
The whole purpose of using `jsonrpsee::core::Error` there is to enable the apps to easier construct the errors that are [`jsonrpsee::core::Error`].

Reply:
gotcha, I think we should break it for ergonomics but let's do it in another PR